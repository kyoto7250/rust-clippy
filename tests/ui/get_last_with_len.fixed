// run-rustfix

#![warn(clippy::get_last_with_len)]

fn dont_use_last() {
    let x = vec![2, 3, 5];
    let _ = x.last(); // ~ERROR Use x.last()

    let y = [2, 3, 5];
    let _ = y.last(); // ~ERROR Use y.last()

    let z = &[2, 3, 5];
    let _ = z.last(); // ~ERROR Use z.last()
}

fn indexing_two_from_end() {
    let x = vec![2, 3, 5];
    let _ = x.get(x.len() - 2);

    let y = [2, 3, 5];
    let _ = y.get(y.len() - 2);

    let z = &[2, 3, 5];
    let _ = z.get(z.len() - 2);
}

fn index_into_last() {
    let x = vec![2, 3, 5];
    let _ = x[x.len() - 1];

    let y = [2, 3, 5];
    let _ = y[y.len() - 1];

    let z = &[2, 3, 5];
    let _ = z[z.len() - 1];
}

fn use_last_with_different_vec_length() {
    let x = vec![2, 3, 5];
    let y = vec!['a', 'b', 'c'];
    let _ = x.get(y.len() - 1);
}

fn main() {
    dont_use_last();
    indexing_two_from_end();
    index_into_last();
    use_last_with_different_vec_length();
}
