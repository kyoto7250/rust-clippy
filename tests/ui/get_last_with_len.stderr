error: accessing last element with `x.get(x.len() - 1)`
  --> $DIR/get_last_with_len.rs:7:13
   |
LL |     let _ = x.get(x.len() - 1); // ~ERROR Use x.last()
   |             ^^^^^^^^^^^^^^^^^^ help: try: `x.last()`
   |
   = note: `-D clippy::get-last-with-len` implied by `-D warnings`

error: accessing last element with `y.get(y.len() - 1)`
  --> $DIR/get_last_with_len.rs:10:13
   |
LL |     let _ = y.get(y.len() - 1); // ~ERROR Use y.last()
   |             ^^^^^^^^^^^^^^^^^^ help: try: `y.last()`

error: accessing last element with `z.get(z.len() - 1)`
  --> $DIR/get_last_with_len.rs:13:13
   |
LL |     let _ = z.get(z.len() - 1); // ~ERROR Use z.last()
   |             ^^^^^^^^^^^^^^^^^^ help: try: `z.last()`

error: aborting due to 3 previous errors

